pkgs/stdenv/linux/default.nix

```diff
@@ -109,12 +106,13 @@ let
     '';
 
 
-  # The bootstrap process proceeds in several steps.
-
-
-  # Create a standard environment by downloading pre-built binaries of
-  # coreutils, GCC, etc.
-
+  # The bootstrap process proceeds in several steps. Our high-level plan
+  # is the following:
+  #
+  # - build binutils with bootstrapTools
+  # - build gcc with bootstrapTools
+  # - build glibc with bootstrapTools + gcc from nixpkgs (embed libgcc_s.so.1)
```

Comment:
@trofi, is there a motivation to rush this through?

Upthread, @K900 [suggested a reason for urgency](https://github.com/NixOS/nixpkgs/pull/209063#discussion_r1063969926), but I doubt that is the motivation for reasons [explained here](https://github.com/NixOS/nixpkgs/pull/209063#discussion_r1063975470) and the fact that there is no mention of aarch64 or `-moutline-atomics` in the commit message.

Are ARMv8.1+ atomics the justification for all the technical debt that this PR incurs?

Reply:
It's not about ARMv8.1+ atomics per se. What's happening here is:

* GCC 11 adds multiple symbols related to aarch64 atomics to libgcc_s, which our bootstrap GCC 9 libgcc_s does not have
* which means applications built with GCC 11 expect those symbols to be there, and crash when they aren't
* which means we can't use GCC 11 on aarch64 without patching things to link the correct libgcc_s
* which is starting to hold back a lot of things that need C++17

For some examples of that, see #209113 (rustc), #201485 (llvm), https://github.com/NixOS/nixpkgs/blob/45ccf597070d2321be4984d2f653e75f5099e4ae/pkgs/desktops/plasma-5/kwin/0001-Revert-x11-Refactor-output-updates.patch (kwin).

"Just update bootstrap-tools" also comes with its own issues as seen in #207135.

I think it's worth building towards a correct solution here, so we don't have to deal with that again.